Add a suffix to the window title when running from a debug build

Calinou opened the pull request:
Since projects exported in debug mode run slower than those exported in release mode, this should be clearly presented to the user.

Thanks @bruvzg for suggesting a [fix for the initial window title](https://github.com/godotengine/godot/issues/20219#issuecomment-547154995) :slightly_smiling_face:

This partially addresses #20219.

## Preview

![DEBUG being appended to the window title](https://user-images.githubusercontent.com/180032/67727053-36e1b000-f9e8-11e9-9649-52582c94f5fb.png)

akien-mga commented:
I would add it only to main but leave the `set_window_title` binding alone, so that users can still override this if they want to.

It might also be worth showing it also on TOOLS enabled builds (but not the editor or project manager) for consistency. Especially if we eventually add an option to run the game from the editor with a release template.

Calinou commented:
> I would add it only to main but leave the set_window_title binding alone, so that users can still override this if they want to.

I made it always visible so that users can notice it even if they override the window title in `_ready()` or a timer (which some kinds of games may be doing, such as idle games). It may look a bit jarring, but that's kind of the point; we want to make sure people don't accidentally distribute debug builds to end users :slightly_smiling_face:

> It might also be worth showing it also on TOOLS enabled builds (but not the editor or project manager) for consistency. Especially if we eventually add an option to run the game from the editor with a release template.

I don't know if this can be done. If the project is running with a release template, it's not running from a tools-enabled build then :slightly_smiling_face: 

That said, we could enable it for editor builds when not running the project from the editor (e.g. when you start the editor from a terminal in the project folder).

akien-mga commented:
> I made it always visible, so that users can notice it even if they override the window title in _ready() or a timer (some kinds of games may be doing this, such as idle games). It may look a bit jarring, but that's kind of the point; we want to make sure developers don't accidentally distribute debug builds to end users.

Sure, but I think some users will actually want to ship debug builds on purpose at some point, and if they need to fork the engine to remove this "(DEBUG)" print, it might be a hurdle. Think about jam games where they might have crashes in release builds and no time to investigate. There are even published console games that were shipped as debug builds back in the days for similar reasons (which is great now to reverse engineer them :P).

Still, I guess we can keep as is and see if anyone complains.

> I don't know if this can be done. If the project is running with a release template, it's not running from a tools-enabled build then

What I meant is that it should show on the debug build you get when you press Run in the editor. *If* we had a feature to test run with the release template, then it would of course not be on the title of the release build, which lets you know at a glance whether you're testing a release or debug build from the editor.

akien-mga commented:
I still think that overriding what `set_window_title()` does when called from scripts is wrong.

IMO the changes in `main.cpp` are sufficient: "DEBUG" will be added to the default window title, and users can choose to override it.

Calinou commented:
@akien-mga That makes sense. I'll change it after I get to resolve the current merge conflicts.

Calinou committed:
Add a suffix to the window title when running from a debug build

Since projects started from the editor or exported in debug mode
run slower than those exported in release mode, this should be
clearly presented to the user.

This partially addresses #20219.

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.2.2.
